contracts/kill_switch/KillSwitch.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity 0.4.24;
+
+import "./IKillSwitch.sol";
+import "./IIssuesRegistry.sol";
+import "../apps/AragonApp.sol";
+import "../common/IsContract.sol";
+
+
+contract KillSwitch is IKillSwitch, IsContract, AragonApp {
+    bytes32 constant public SET_DEFAULT_ISSUES_REGISTRY_ROLE = keccak256("SET_DEFAULT_ISSUES_REGISTRY_ROLE");
+    bytes32 constant public SET_ISSUES_REGISTRY_ROLE = keccak256("SET_ISSUES_REGISTRY_ROLE");
+    bytes32 constant public SET_CONTRACT_ACTION_ROLE = keccak256("SET_CONTRACT_ACTION_ROLE");
+    bytes32 constant public SET_HIGHEST_ALLOWED_SEVERITY_ROLE = keccak256("SET_HIGHEST_ALLOWED_SEVERITY_ROLE");
+
+    string constant private ERROR_ISSUES_REGISTRY_NOT_CONTRACT = "KS_ISSUES_REGISTRY_NOT_CONTRACT";
+
+    enum ContractAction { Check, Ignore, Deny }
+
+    struct Settings {
+        ContractAction action;
+        IIssuesRegistry.Severity highestAllowedSeverity;
+        IIssuesRegistry issuesRegistry;
+    }
+
+    IIssuesRegistry public defaultIssuesRegistry;
+    mapping (address => Settings) internal contractSettings;
+
+    event DefaultIssuesRegistrySet(address issuesRegistry);
+    event ContractActionSet(address indexed contractAddress, ContractAction action);
+    event IssuesRegistrySet(address indexed contractAddress, address issuesRegistry);
+    event HighestAllowedSeveritySet(address indexed contractAddress, IIssuesRegistry.Severity severity);
+
+    function initialize(IIssuesRegistry _defaultIssuesRegistry) external onlyInit {
+        initialized();
+        _setDefaultIssuesRegistry(_defaultIssuesRegistry);
+    }
+
+    function setDefaultIssuesRegistry(IIssuesRegistry _defaultIssuesRegistry)
+        external
+        authP(SET_DEFAULT_ISSUES_REGISTRY_ROLE, arr(msg.sender))
+    {
+        _setDefaultIssuesRegistry(_defaultIssuesRegistry);
+    }
+
+    function setContractAction(address _contract, ContractAction _action)
+        external
+        authP(SET_CONTRACT_ACTION_ROLE, arr(_contract, msg.sender))
+    {
+        contractSettings[_contract].action = _action;
+        emit ContractActionSet(_contract, _action);
+    }
+
+    function setHighestAllowedSeverity(address _contract, IIssuesRegistry.Severity _severity)
+        external
+        authP(SET_HIGHEST_ALLOWED_SEVERITY_ROLE, arr(_contract, msg.sender))
+    {
+        contractSettings[_contract].highestAllowedSeverity = _severity;
+        emit HighestAllowedSeveritySet(_contract, _severity);
+    }
+
+    function setIssuesRegistry(address _contract, IIssuesRegistry _issuesRegistry)
+        external
+        authP(SET_ISSUES_REGISTRY_ROLE, arr(_contract, msg.sender))
+    {
+        require(isContract(_issuesRegistry), ERROR_ISSUES_REGISTRY_NOT_CONTRACT);
+        contractSettings[_contract].issuesRegistry = _issuesRegistry;
+        emit IssuesRegistrySet(_contract, address(_issuesRegistry));
+    }
+
+    function shouldDenyCallingContract(address _contract) external returns (bool) {
```

Comment:
It would be nice to have an estimation of how much gas overload means adding the modifier that calls this to a function.